SEImplementation/SEImplementation/Plugin/CoreThresholdPartition/CoreThresholdPartitionTask.h

```diff
@@ -0,0 +1,72 @@
+/** Copyright © 2019 Université de Genève, LMU Munich - Faculty of Physics, IAP-CNRS/Sorbonne Université
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 3.0 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+/**
+ * @file CoreThresholdPartitionTask.h
+ *
+ * @date May 27, 2019
+ * @author mkuemmel@usm.lmu.de
+ */
+
+#ifndef _SEIMPLEMENTATION_PLUGIN_CORETHRESHOLDTASK_H_
+#define _SEIMPLEMENTATION_PLUGIN_CORETHRESHOLDTASK_H_
+
+#include "NCorePixel.h"
+#include "SEFramework/Property/DetectionFrame.h"
+#include "SEImplementation/Property/PixelCoordinateList.h"
+
+#include "SEFramework/Task/SourceTask.h"
+#include "SEImplementation/Plugin/DetectionFramePixelValues/DetectionFramePixelValues.h"
+
+#include "SEImplementation/Measurement/MultithreadedMeasurement.h"
+
+namespace SourceXtractor {
+
+class CoreThresholdPartitionTask : public SourceTask {
+
+public:
+  CoreThresholdPartitionTask(double snr_level): m_snr_level(snr_level) {};
+
+  virtual ~CoreThresholdPartitionTask() = default;
+
+  virtual void computeProperties(SourceInterface& source) const {
+    std::lock_guard<std::recursive_mutex> lock(MultithreadedMeasurement::g_global_mutex);
+
+    long int n_snr_level=0;
+
+    // get the detection frame and the SNR image
+    const auto& snr_image    = source.getProperty<DetectionFrame>().getFrame()->getSnrImage();
+
+    // go over all pixels
+    for (auto pixel_coord : source.getProperty<PixelCoordinateList>().getCoordinateList())
```

Comment:
I'd put brackets around the code inside the for. Not having them, and because of the indent of "n_snr_leven += 1" I got confused for a moment.